doc/src/modules/vector/coordsys.rst

```diff
@@ -0,0 +1,273 @@
+in its API-
+
+1. ``orient_new_axis``
+
+2. ``orient_new_body``
+
+3. ``orient_new_space``
+
+4. ``orient_new_quaternion``
+
+Please look at the ``CoordSysCartesian`` class API given in the docs
+of this module, to know their functionality and required arguments 
+in detail.
+
+Using Orienter(s) and the orient_new method
+-------------------------------------------
+
+Suppose you want to rotate the new system about the axis
+:math:`\mathbf{\hat{k}}`, by angle :math:`a`.
+
+You would first have to initialize an ``AxisOrienter`` instance for 
+storing the rotation information.
+
+  >>> from sympy.vector import CoordSysCartesian, AxisOrienter
+  >>> from sympy.abc import a
+  >>> N = CoordSysCartesian('N')
+  >>> axis_orienter = AxisOrienter(a, N.k)
+
+And then apply it using the ``orient_new`` method.
+
+  >>> M = N.orient_new('M', axis_orienter)
+
+``orient_new`` also lets you orient new systems using multiple
+``Orienter`` instances, provided in an iterable. The rotations/orientations
+are applied to the new system in the order the ``Orienter`` instances 
+appear in the iterable.
+
+  >>> from sympy.vector import BodyOrienter
+  >>> from sympy.abc import a, b, c
+  >>> body_orienter = BodyOrienter(a, b, c, 'XYZ')
+  >>> M = N.orient_new('M', (axis_orienter, body_orienter))
+
+The :mod:`sympy.vector` API provides the following four ``Orienter``
+classes for orientation purposes-
+
+1. ``AxisOrienter``
+
+2. ``BodyOrienter``
+
+3. ``SpaceOrienter``
+
+4. ``QuaternionOrienter``
+
+Please refer to the API of the respective classes in the docs of this
+module to know more.
+
+
+In each of the above examples, the origin of the new coordinate system
+coincides with the origin of the 'parent' system.
+
+  >>> M.position_wrt(N)
+  0
+
+To compute the rotation matrix of any coordinate system with respect 
+to another one, use the ``rotation_matrix`` method.
+
+  >>> M = N.orient_new_axis('M', a, N.k)
+  >>> M.rotation_matrix(N)
+  Matrix([
+  [ cos(a), sin(a), 0],
+  [-sin(a), cos(a), 0],
+  [      0,      0, 1]])
+  >>> M.rotation_matrix(M)
+  Matrix([
+  [1, 0, 0],
+  [0, 1, 0],
+  [0, 0, 1]])
+  
+
+Orienting AND Locating new systems
+==================================
+
+What if you want to initialize a new system that is not only oriented
+in a pre-defined way, but also translated with respect to the parent?
+
+Each of the ``orient_new_<method of orientation>`` methods, as well
+as the ``orient_new`` method, support a ``location`` keyword
+argument.
+
+If a ``Vector`` is supplied as the value for this ``kwarg``, the
+new system's origin is automatically defined to be located at that
+position vector with respect to the parent coordinate system.
+
+Thus, the orientation methods also act as methods to support orientation+
+location of the new systems.
+
+  >>> M = N.orient_new_axis('M', a, N.k, location=2*N.j)
+  >>> M.position_wrt(N)
+  2*N.j
+  >>> from sympy.vector import express
+  >>> express(N.position_wrt(M), M)
```

Comment:
Will add this. However, using the express function is more general, since it can re-express any quantity (scalars/vectors/dyadics) in a given system.